opennms-config/src/main/java/org/opennms/netmgt/config/DiscoveryConfigFactory.java

```diff
@@ -486,6 +596,55 @@ public String getForeignSource(InetAddress address) {
         }
     }
 
+    public List<Detector> getListOfDetectors(InetAddress inetAddress, String location) {
+
+        String ipAddress = InetAddressUtils.toIpAddrString(inetAddress);
+        if (Strings.isNullOrEmpty(location)) {
+            location = LocationUtils.DEFAULT_LOCATION_NAME;
+        }
+        final String sourceLocation = location;
+        final String defaultLocation = getConfiguration().getLocation().orElse(LocationUtils.DEFAULT_LOCATION_NAME);
+
+        // Find the definition that has the IP Address and return the detectors for that definition.
+        Optional<Definition> definition = getConfiguration().getDefinitions().stream().filter(def -> {
+
+            //First match should be enough to pass filter.
+            boolean specificMatch = def.getSpecifics().stream().anyMatch(specific -> {
+                return specific.getLocation().orElse(def.getLocation().orElse(defaultLocation)).equals(sourceLocation)
+                        && specific.getAddress().equals(ipAddress);
+            });
+            // Find match in url specifics
+            if (!specificMatch) {
```

Comment:
We can return true immediately if `specificMatch` is true and avoid indenting this block.